### PR TITLE
Add support for dynamic workload type

### PR DESCRIPTION
**Pull request by preetha-intel**
### Description
Support to set EPdynamic options in OVEP 

### Motivation and Context
relate to [https://github.com/microsoft/onnxruntime/pull/22282.](https://github.com/microsoft/onnxruntime/pull/22282.)

### CODE REVIEW

**Comment by sfatimar** [comment]
Please raise this branch to ovep-develop-lnl-1.4

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc [hunk 1]
```diff
@@ -198,4 +198,13 @@ std::vector<AllocatorPtr> OpenVINOExecutionProvider::CreatePreferredAllocators()
 }
 #endif
 
+common::Status OpenVINOExecutionProvider::SetEpDynamicOptions(gsl::span<const char* const> keys,
+                                                              gsl::span<const char* const> values) {
+  if ((std::find(keys.begin(), keys.end(), "ep.dynamic.workload_type") != keys.end()) &&
```
Comment (MayureshV1) [inline onnxruntime/core/providers/openvino/openvino_execution_provider.cc:203]: Also needs implementation for Setting Dynamic Option -> ep.dynamic.workload_type to "DEFAULT"

File: onnxruntime/core/providers/openvino/backend_manager.cc [hunk 2]
```diff
@@ -502,6 +502,12 @@ void BackendManager::Compute(OrtKernelContext* context) {
 
     dynamic_backend->Infer(context);
   } else {
+    if (dynamic_workload_type == "EFFICIENT") {
```
Comment (MayureshV1) [inline onnxruntime/core/providers/openvino/backend_manager.cc:505]: Also needs implementation for Setting Dynamic Option -> ep.dynamic.workload_type to "DEFAULT"

Comment (preetha-intel) [reply onnxruntime/core/providers/openvino/backend_manager.cc:505]: Done

File: onnxruntime/core/providers/openvino/openvino_execution_provider.cc (hunk 1 above)
Comment (preetha-intel) [reply onnxruntime/core/providers/openvino/openvino_execution_provider.cc:203]: Added the changes

**Comment by sfatimar** [comment]
Please change PR to ovep-develop-lnl-1.4

**Comment by preetha-intel** [comment]
Closing this PR as the base branch is frozen.

**Pull request closed without merging**